### PR TITLE
Updated docstring links 

### PR DESCRIPTION
**Pull request by XFarooqi**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?

This Pull Request will Fix issue #504 and also I replace all the links in Codebase.

If anything needs to be correct, Let me Know!

## Did you have fun?

Yeah, I had fun finding the links in the docstrings and placing them in links.rst.

### CODE REVIEW

**Comment by Borda** [comment]
@XFarooqi pls don't open any more PRs on the same topic

**Comment by XFarooqi** [comment]
> @XFarooqi pls don't open any more PRs on the same topic

I am sorry, I have removed previous ones.

File: docs/source/links.rst [hunk 1]
```diff
@@ -37,6 +36,37 @@
 .. _IR Fall-out: https://en.wikipedia.org/wiki/Evaluation_measures_(information_retrieval)#Fall-out
 .. _MAPE implementation returns: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_absolute_percentage_error.html
 .. _mean squared logarithmic error: https://scikit-learn.org/stable/modules/model_evaluation.html#mean-squared-log-error
-.. _Mean Reciprocal Rank: https://en.wikipedia.org/wiki/Mean_reciprocal_rank
 .. _LPIPS: https://arxiv.org/abs/1801.03924
 .. _Tweedie Deviance Score: https://en.wikipedia.org/wiki/Tweedie_distribution#The_Tweedie_deviance
+.. _Permutation Invariant Training of Deep Models: https://ieeexplore.ieee.org/document/7952154
```
Comment (Borda) [inline docs/source/links.rst:41]: I may suggest adding one by one where each commit will be moving one link here to this aggregating list and replacing it accordingly in the codebase so it is much easier to find which changes is breaking it...

**Review by Borda — COMMENTED** [review]
mind pls check the failing docs tests? :rabbit:

File: torchmetrics/text/bert.py [hunk 2]
```diff
@@ -38,12 +38,12 @@ def _concatenate(d: Dict[str, List[torch.Tensor]]) -> Dict[str, torch.Tensor]:
 
 
 class BERTScore(Metric):
-    """`BERTScore <https://arxiv.org/abs/1904.09675>`_ leverages the pre-trained contextual embeddings from BERT
-    and matches words in candidate and reference sentences by cosine similarity. It has been shown to correlate
-    with human judgment on sentence-level and system-level evaluation. Moreover, BERTScore computes precision,
-    recall, and F1 measure, which can be useful for evaluating different language generation tasks.
+    """`Bert_score Evaluating Text Generation`_ leverages the pre-trained contextual embeddings from BERT and
```
Comment (SkafteNicki) [inline torchmetrics/text/bert.py:41]:
```suggestion
    """
    `Bert_score Evaluating Text Generation`_ leverages the pre-trained contextual embeddings from BERT and
```

**Review by SkafteNicki — APPROVED** [review]

File: torchmetrics/functional/classification/accuracy.py [hunk 3]
```diff
@@ -265,7 +265,7 @@ def accuracy(
     multiclass: Optional[bool] = None,
     ignore_index: Optional[int] = None,
 ) -> Tensor:
-    r"""Computes `Accuracy <https://en.wikipedia.org/wiki/Accuracy_and_precision>`_:
+    r"""`Accuracy`_
```
Comment (Borda) [inline torchmetrics/functional/classification/accuracy.py:268]:
```suggestion
    r"""Computes `Accuracy`_
```

File: torchmetrics/functional/classification/cohen_kappa.py [hunk 4]
```diff
@@ -74,39 +74,39 @@ def cohen_kappa(
     threshold: float = 0.5,
 ) -> Tensor:
     r"""
-    Calculates `Cohen's kappa score <https://en.wikipedia.org/wiki/Cohen%27s_kappa>`_ that measures
-    inter-annotator agreement. It is defined as
+    `Cohen's kappa score`_
```
Comment (Borda) [inline torchmetrics/functional/classification/cohen_kappa.py:77]:
```suggestion
    Calculates `Cohen's kappa score`_ that measures inter-annotator agreement.
```

File: torchmetrics/functional/classification/iou.py [hunk 5]
```diff
@@ -76,7 +76,7 @@ def iou(
     reduction: str = "elementwise_mean",
 ) -> Tensor:
     r"""
-    Computes `Intersection over union, or Jaccard index calculation <https://en.wikipedia.org/wiki/Jaccard_index>`_:
+    `Jaccard index`_
```
Comment (Borda) [inline torchmetrics/functional/classification/iou.py:79]:
```suggestion
    Computes `Jaccard index`_
```

File: torchmetrics/functional/classification/kl_divergence.py [hunk 6]
```diff
@@ -79,7 +79,7 @@ def _kld_compute(measures: Tensor, total: Tensor, reduction: Optional[str] = "me
 
 
 def kl_divergence(p: Tensor, q: Tensor, log_prob: bool = False, reduction: Optional[str] = "mean") -> Tensor:
-    r"""Computes the `KL divergence <https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence>`_:
+    r"""`KL divergence`_
```
Comment (Borda) [inline torchmetrics/functional/classification/kl_divergence.py:82]:
```suggestion
    r"""Computes `KL divergence`_
```

File: torchmetrics/functional/classification/precision_recall.py [hunk 7]
```diff
@@ -84,7 +84,7 @@ def precision(
     multiclass: Optional[bool] = None,
 ) -> Tensor:
     r"""
-    Computes `Precision <https://en.wikipedia.org/wiki/Precision_and_recall>`_:
+    `Precision`_
```
Comment (Borda) [inline torchmetrics/functional/classification/precision_recall.py:87]:
```suggestion
    Computes `Precision`_
```

File: torchmetrics/functional/classification/precision_recall.py [hunk 8]
```diff
@@ -281,7 +281,7 @@ def recall(
     multiclass: Optional[bool] = None,
 ) -> Tensor:
     r"""
-    Computes `Recall <https://en.wikipedia.org/wiki/Precision_and_recall>`_:
+    `Recall`_
```
Comment (Borda) [inline torchmetrics/functional/classification/precision_recall.py:284]:
```suggestion
    Computes `Recall`_
```

File: torchmetrics/functional/classification/specificity.py [hunk 9]
```diff
@@ -79,7 +79,7 @@ def specificity(
     multiclass: Optional[bool] = None,
 ) -> Tensor:
     r"""
-    Computes `Specificity <https://en.wikipedia.org/wiki/Sensitivity_and_specificity>`_:
+    `Specificity`_
```
Comment (Borda) [inline torchmetrics/functional/classification/specificity.py:82]:
```suggestion
    Computes `Specificity`_
```

File: torchmetrics/functional/image/gradients.py [hunk 10]
```diff
@@ -46,8 +46,7 @@ def _compute_image_gradients(img: Tensor) -> Tuple[Tensor, Tensor]:
 
 
 def image_gradients(img: Tensor) -> Tuple[Tensor, Tensor]:
-    """Computes the `gradients <https://en.wikipedia.org/wiki/Image_gradient>`_ of a given image using finite
-    difference.
+    """`Gradient Computation of Image`_
```
Comment (Borda) [inline torchmetrics/functional/image/gradients.py:49]:
```suggestion
    """Computes `Gradient Computation of Image`_ of a given image using finite difference.
```

File: torchmetrics/functional/nlp.py [hunk 11]
```diff
@@ -25,8 +25,7 @@ def bleu_score(
     n_gram: int = 4,
     smooth: bool = False,
 ) -> Tensor:
-    """Calculate `BLEU score <https://en.wikipedia.org/wiki/BLEU>`_ of machine translated text with one or more
-    references.
+    """`BLEU score`_
```
Comment (Borda) [inline torchmetrics/functional/nlp.py:28]:
```suggestion
    """Calculate `BLEU score`_ of machine translated text with one or more references.
```

File: torchmetrics/functional/regression/r2score.py [hunk 12]
```diff
@@ -26,8 +26,7 @@ def r2score(
     multioutput: str = "uniform_average",
 ) -> Tensor:
     r"""
-    Computes r2 score also known as `coefficient of determination
-    <https://en.wikipedia.org/wiki/Coefficient_of_determination>`_:
+    `R2 Score_Coefficient Determination`_
```
Comment (Borda) [inline torchmetrics/functional/regression/r2score.py:29]:
```suggestion
    Computes r2 score also known as `R2 Score_Coefficient Determination`_
```

File: torchmetrics/functional/text/bert.py [hunk 13]
```diff
@@ -376,7 +376,7 @@ def _get_precision_recall_f1(
 
 
 def _get_hash(model_name_or_path: Optional[str] = None, num_layers: Optional[int] = None, idf: bool = False) -> str:
-    """Copied from https://github.com/Tiiiger/bert_score/blob/master/bert_score/utils.py and adjusted."""
+    """`BERT_score`_"""
```
Comment (Borda) [inline torchmetrics/functional/text/bert.py:379]:
```suggestion
    """Compute `BERT_score`_ (copied and adjusted)"""
```

File: torchmetrics/functional/text/rouge.py [hunk 14]
```diff
@@ -285,7 +285,7 @@ def rouge_score(
             If any of the ``rouge_keys`` does not belong to the allowed set of keys.
 
     References:
-        [1] ROUGE: A Package for Automatic Evaluation of Summaries by Chin-Yew Lin https://aclanthology.org/W04-1013/
+        [1] ROUGE: A Package for Automatic Evaluation of Summaries by Chin-Yew Lin. `Rouge Detail`_
```
Comment (Borda) [inline torchmetrics/functional/text/rouge.py:288]:
```suggestion
        [1] ROUGE: A Package for Automatic Evaluation of Summaries by Chin-Yew Lin. https://aclanthology.org/W04-1013/
```

File: torchmetrics/image/inception.py [hunk 15]
```diff
@@ -74,11 +74,11 @@ class IS(Metric):
     References:
         [1] Improved Techniques for Training GANs
         Tim Salimans, Ian Goodfellow, Wojciech Zaremba, Vicki Cheung, Alec Radford, Xi Chen
-        https://arxiv.org/abs/1606.03498
+        `Improved Techniques for Training GANs`_
```
Comment (Borda) [inline torchmetrics/image/inception.py:77]:
```suggestion
        https://arxiv.org/abs/1606.03498
```

File: torchmetrics/image/inception.py [hunk 16]
```diff
@@ -74,11 +74,11 @@ class IS(Metric):
...
 
         [2] GANs Trained by a Two Time-Scale Update Rule Converge to a Local Nash Equilibrium,
         Martin Heusel, Hubert Ramsauer, Thomas Unterthiner, Bernhard Nessler, Sepp Hochreiter
-        https://arxiv.org/abs/1706.08500
+        `GANs Trained by a Two Time-Scale`_
```
Comment (Borda) [inline torchmetrics/image/inception.py:81]:
```suggestion
        https://arxiv.org/abs/1706.08500
```

File: torchmetrics/image/kid.py [hunk 17]
```diff
@@ -25,7 +25,7 @@
 
 
 def maximum_mean_discrepancy(k_xx: Tensor, k_xy: Tensor, k_yy: Tensor) -> Tensor:
-    """Adapted from https://github.com/toshas/torch- fidelity/blob/v0.3.0/torch_fidelity/metric_kid.py."""
+    """`KID Score`_"""
```
Comment (Borda) [inline torchmetrics/image/kid.py:28]:
```suggestion
    """Adapted from `KID Score`_"""
```

File: torchmetrics/image/kid.py [hunk 18]
```diff
@@ -45,7 +45,7 @@ def maximum_mean_discrepancy(k_xx: Tensor, k_xy: Tensor, k_yy: Tensor) -> Tensor
 
 
 def poly_kernel(f1: Tensor, f2: Tensor, degree: int = 3, gamma: Optional[float] = None, coef: float = 1.0) -> Tensor:
-    """Adapted from https://github.com/toshas/torch- fidelity/blob/v0.3.0/torch_fidelity/metric_kid.py."""
+    """`KID Score`_"""
```
Comment (Borda) [inline torchmetrics/image/kid.py:48]:
```suggestion
    """Adapted from `KID Score`_"""
```

File: torchmetrics/image/kid.py [hunk 19]
```diff
@@ -125,11 +125,11 @@ class KID(Metric):
     References:
         [1] Demystifying MMD GANs
         Mikołaj Bińkowski, Danica J. Sutherland, Michael Arbel, Arthur Gretton
-        https://arxiv.org/abs/1801.01401
+        `Demystifying MMD GANs`_
```
Comment (Borda) [inline torchmetrics/image/kid.py:128]:
```suggestion
        https://arxiv.org/abs/1801.01401
```

File: torchmetrics/image/kid.py [hunk 20]
```diff
@@ -125,11 +125,11 @@ class KID(Metric):
...
 
         [2] GANs Trained by a Two Time-Scale Update Rule Converge to a Local Nash Equilibrium,
         Martin Heusel, Hubert Ramsauer, Thomas Unterthiner, Bernhard Nessler, Sepp Hochreiter
-        https://arxiv.org/abs/1706.08500
+        `GANs Trained by a Two Time-Scale`_
```
Comment (Borda) [inline torchmetrics/image/kid.py:132]:
```suggestion
        https://arxiv.org/abs/1706.08500
```

File: torchmetrics/image/kid.py [hunk 21]
```diff
@@ -252,7 +252,7 @@ def compute(self) -> Tuple[Tensor, Tensor]:
         """Calculate KID score based on accumulated extracted features from the two distributions. Returns a tuple
         of mean and standard deviation of KID scores calculated on subsets of extracted features.
 
-        Implementation inspired by https://github.com/toshas/torch-fidelity/blob/v0.3.0/torch_fidelity/metric_kid.py
+        `Fid Score`_
```
Comment (Borda) [inline torchmetrics/image/kid.py:255]:
```suggestion
        Implementation inspired by `Fid Score`_
```

**Review by Borda — APPROVED** [review]
